source/includes/code-examples/GridFS.cs

```diff
@@ -0,0 +1,415 @@
+using MongoDB.Bson;
+using MongoDB.Driver;
+using MongoDB.Driver.GridFS;
+
+namespace Fundamentals;
+
+class GridFS
+{
+    static void CreateBucket()
+    {
+        // Initialize MongoDB client
+        // start-create-bucket
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+        // end-create-bucket
+    }
+
+    static void CreateCustomBucket()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates or references a GridFS bucket with a custom name
+        // start-create-custom-bucket
+        var options = new GridFSBucketOptions { BucketName = "myCustomBucket" };
+        var customBucket = new GridFSBucket(database, options);
+        // end-create-custom-bucket
+    }
+
+    static async Task UploadFileAsync()
+    {
+        // Initialize MongoDB client
+        var client = new MongoClient("<connection string>");
+        var database = client.GetDatabase("db");
+
+        // Creates a GridFS bucket or references an existing one
+        var bucket = new GridFSBucket(database);
+
+        // Uploads a file called "my_file" to the GridFS bucket and writes data to it
+        // start-open-upload-stream-async
+        using (var uploader = await bucket.OpenUploadStreamAsync("my_file", options))
+        {
+            // ASCII for "HelloWorld"
+            byte[] bytes = { 72, 101, 108, 108, 111, 87, 111, 114, 108, 100 };
+
+            for (int i = 0; i < 5; i++)
+            {
+                await uploader.WriteAsync(bytes, 0, bytes.Length);
+            }
```

Comment:
I don't understand this example. What are we doing? What's the point of writing the same 10 bytes 5 times?

Reply:
Just to put some data in the file. This example was taken from an [example in the C++ driver.](https://github.com/mongodb/mongo-cxx-driver/blob/e6e2696826a5c64e87c76bf31673dcf16aefc83f/examples/mongocxx/gridfs.cpp)

Reply:
OK... but as a reader I wasted time trying to figure out why we were inserting the same bytes 5 times.

Why not just upload the bytes once? Doing it 5 times adds nothing to the example and causes confusion because the reader is trying to figure out what the significance of doing it 5 times is.

Reply:
We can do that!